src/Uno.UI/UI/Xaml/Controls/Repeater/Phaser.cs

```diff
@@ -0,0 +1,240 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Windows.Foundation;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls.Primitives;
+using Windows.UI.Xaml.Markup;
+using Uno.UI.Helpers.WinUI;
+
+namespace Microsoft.UI.Xaml.Controls
+{
+	internal class Phaser
+	{
+		private readonly ItemsRepeater m_owner;
+		private List<ElementInfo> m_pendingElements = new List<ElementInfo>();
+		private bool m_registeredForCallback;
+
+		public Phaser(ItemsRepeater owner)
+		{
+			// ItemsRepeater is not fully constructed yet. Don't interact with it.
+
+			m_owner = owner;
+		}
+
+		public void PhaseElement(UIElement element, VirtualizationInfo virtInfo)
+		{
+			var dataTemplateComponent = virtInfo.DataTemplateComponent;
+			var nextPhase = virtInfo.Phase;
+			bool shouldPhase = false;
+
+			if (nextPhase > 0)
+			{
+				if (dataTemplateComponent != null)
+				{
+					// Perf optimization: RecyclingElementFactory sets up the virtualization info so we don't need to update it here.
+					shouldPhase = true;
+				}
+				else
+				{
+					throw new InvalidOperationException("Phase was set on virtualization info, but dataTemplateComponent was not.");
+				}
+			}
+			else if (nextPhase == VirtualizationInfo.PhaseNotSpecified)
+			{
+				// If virtInfo.Phase is not specified, virtInfo.DataTemplateComponent cannot be valid.
+				global::System.Diagnostics.Debug.Assert(dataTemplateComponent == null);
+				// ItemsRepeater might be using a custom view generator in which case, virtInfo would not be bootstrapped.
+				// In this case, fallback to querying for the data template component and setup virtualization info now.
+				dataTemplateComponent = XamlBindingHelper.GetDataTemplateComponent(element);
+				if (dataTemplateComponent != null)
+				{
+					// Clear out old data. 
+					dataTemplateComponent.Recycle();
+
+					nextPhase = VirtualizationInfo.PhaseReachedEnd;
```

Comment:
Codacy found an issue: [Remove this useless assignment to local variable 'nextPhase'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5863680)